python/GafferSceneUI/LightEditor.py

```diff
@@ -354,6 +382,120 @@ def __editSelectedCells( self, pathListing ) :
+			for inspector, pathInspections in inspectors.items() :
+				for path, inspection in pathInspections.items() :
+					plug = inspection.acquireEdit()
+
+					if isinstance( plug, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) :
+						plug["value"].setValue( newValue )
+						plug["enabled"].setValue( True )
+						if isinstance( plug, Gaffer.TweakPlug ) :
+							plug["mode"].setValue( Gaffer.TweakPlug.Mode.Create )
+					else :
+						plug.setValue( newValue )
+
+		return True
+
+	def __activeInspectionTweaks( self, pathListing ) :
+
+		tweaks = []
+
+		with Gaffer.Context( self.getContext() ) as context :
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if not isinstance( column, _GafferSceneUI._LightEditorInspectorColumn ) :
+					continue
+				for path in columnSelection.paths() :
+					context["scene:path"] = GafferScene.ScenePlug.stringToPath( path )
+					inspection = column.inspector().inspect()
+					if inspection is not None and inspection.editable() :
+						source = inspection.source()
+						editScope = self.__settingsNode["editScope"].getInput()
+						if (
+							isinstance( source, ( Gaffer.TweakPlug, Gaffer.NameValuePlug ) ) and
+							( editScope is None or editScope.node().isAncestorOf( source ) ) and
+							source["enabled"].getValue()
+						) :
+							tweaks.append( source )
+
+		return tweaks
+
+	def __removableAttributeInspections( self, pathListing ) :
+
+		inspections = []
+
+		with Gaffer.Context( self.getContext() ) as context :
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if (
+					not isinstance( column, _GafferSceneUI._LightEditorInspectorColumn ) or
+					not isinstance( column.inspector(), GafferSceneUI.Private.AttributeInspector )
+				) :
+					continue
+				for path in columnSelection.paths() :
+					context["scene:path"] = GafferScene.ScenePlug.stringToPath( path )
+					inspection = column.inspector().inspect()
+					if inspection is not None and inspection.editable() :
+						source = inspection.source()
+						editScope = self.__settingsNode["editScope"].getInput()
+						if (
+							( isinstance( source, Gaffer.TweakPlug ) and source["mode"].getValue() != Gaffer.TweakPlug.Mode.Remove ) or
+							editScope is not None
+						) :
+							inspections.append( inspection )
+
+		return inspections
+
+	def __disableEdits( self, pathListing ) :
+
+		tweaks = self.__activeInspectionTweaks( pathListing )
+
+		with Gaffer.UndoScope( self.scriptNode() ) :
+			for tweak in tweaks :
+				tweak["enabled"].setValue( False )
+
+	def __removeAttributes( self, pathListing ) :
+
+		inspections = self.__removableAttributeInspections( pathListing )
+
+		with Gaffer.UndoScope( self.scriptNode() ) :
+			for inspection in inspections :
+				tweak = inspection.acquireEdit()
+				tweak["enabled"].setValue( True )
+				tweak["mode"].setValue( Gaffer.TweakPlug.Mode.Remove )
```

Comment:
Once we've removed an attribute, viewing the attribute history gives this warning : ```PathListingWidget : Could not find file "removeSmall.png"```.

Reply:
Added the dotted-line box to indicate removal in https://github.com/GafferHQ/gaffer/pull/4948/commits/a28d7e853342e0260c8afa2f90b3be41392b4fb2